doc: add the 2025.1 upgrade guides and reorganize the upgrade section (#22352)
scylladb/scylladb: docs/upgrade/about-upgrade.rst

================
About Upgrade
================

> ScyllaDB upgrade is a rolling procedure - it does not require a full cluster

Reviewer: ```suggestion
ScyllaDB upgrade is a rolling procedure - it is performed without any downtime or disruption of service
```

(remove the 'does not require a full cluster shutdown')

shutdown and is performed without any downtime or disruption of service.

To ensure a successful upgrade, follow
the :doc:`documented upgrade procedures <upgrade-guides/index>` tested by
ScyllaDB. This means that:

* You should perform the upgrades consecutively - to each successive X.Y

>   version, **without skipping any major or minor version**.

Reviewer: I thought we do allow 2024.1 -> 2025.1 (skipping 2024.2)?

> * Before you upgrade to the next version, the whole cluster (each node) must

Reviewer: ```suggestion
* Before you upgrade to the next version, the whole cluster (every node) must
```

  be upgraded to the previous version.
* You cannot perform an upgrade by replacing the nodes in the cluster with new
  nodes with a different ScyllaDB version. You should never add a new node with
  a different version to a cluster - if you
  :doc:`add a node </operating-scylla/procedures/cluster-management/add-node-to-cluster>`,
  it must have the same X.Y.Z (major.minor.patch) version as the other nodes in
  the cluster.

Upgrading to each patch version by following the Maintenance Release Upgrade

> Guide is optional. However, we recommend upgrading to the latest patch release

Reviewer: ```suggestion
Guide is optional. We highly recommend upgrading to the latest patch release
```
